src/snapred/backend/data/LocalDataService.py

```diff
@@ -1258,3 +1259,97 @@ def writePixelMask(
+
+    ## LIVE-DATA SUPPORT METHODS
+
+    def hasLiveDataConnection(self) -> bool:
+        """For 'live data' methods: test if there is a listener connection to the instrument."""
+
+        # In addition to 'analysis.sns.gov', other nodes on the subnet should be OK as well.
+        #   So this check should also return True on those nodes.
+        # If this method returns True, then the `SNSLiveEventDataListener` should be able to function.
+
+        # Normalize to an actual "URL" and then strip off the protocol (not actually "http") and port:
+        #   `liveDataAddress` returns a string similar to "bl3-daq1.sns.gov:31415".
+
+        facility, instrument = Config["liveData.facility.name"], Config["liveData.instrument.name"]
+        hostname = urlparse(
+            "http://" + ConfigService.getFacility(facility).instrument(instrument).liveDataAddress()
+        ).hostname
+        status = True
+        try:
+            socket.gethostbyaddr(hostname)
+        except Exception:  # noqa: BLE001
+            # specifically:
+            #   we're expecting a `socket.gaierror`, but any exception will indicate that there's no connection
+            status = False
+        return status
+
+    def _liveMetadataFromRun(self, run: Run) -> LiveMetadata:
+        """Construct a 'LiveMetadata' instance from a 'mantid.api.Run' instance."""
+
+        logs = mappingFromRun(run)
+        metadata = None
+        try:
+            run_number: str = str(logs["run_number"])
+
+            # See comment at `snapred.backend.data.util.PV_logs_util.mappingFromRun` about this conversion.
+            start_time: datetime.datetime = datetime.datetime.replace(
+                np.datetime64(logs["start_time"], "us").astype(datetime.datetime), tzinfo=datetime.timezone.utc
+            )
+            end_time: datetime.datetime = datetime.datetime.replace(
+                np.datetime64(logs["end_time"], "us").astype(datetime.datetime), tzinfo=datetime.timezone.utc
+            )
+
+            # Many required log values will not be present if a run is inactive.
+            detector_state = (
+                self._detectorStateFromMapping(logs) if run_number != str(LiveMetadata.INACTIVE_RUN) else None
+            )
+
+            proton_charge = logs["proton_charge"]
+
+            metadata = LiveMetadata(
+                runNumber=run_number,
+                startTime=start_time,
+                endTime=end_time,
+                detectorState=detector_state,
+                protonCharge=proton_charge,
+            )
+        except (KeyError, RuntimeError, ValidationError) as e:
+            raise RuntimeError("unable to extract LiveMetadata from Run") from e
+        return metadata
+
+    def _readLiveData(self, ws: WorkspaceName, duration: int):
+        # 'StartTime=""' => read all of the available data
+
+        startTime = (
+            (datetime.datetime.utcnow() + datetime.timedelta(seconds=-duration)).isoformat() if duration != 0 else ""
```

Comment:
Comment to clarify this is `duration` seconds in the passed, just so its easier to glance at.